libs/clients/health-directorate/src/lib/healthDirectorateClient.service.ts

```diff
@@ -1,17 +1,112 @@
 import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
-import { MinarSidur, StarfsleyfiAMinumSidumApi } from '../../gen/fetch'
+import {
+  MinarSidur,
+  StarfsleyfiAMinumSidumApi,
+  VottordApi,
+} from '../../gen/fetch'
 import { Injectable } from '@nestjs/common'
+import {
+  HealthcareLicense,
+  HealthcareLicenseCertificate,
+  HealthcareLicenseCertificateRequest,
+} from './healthDirectorateClient.types'
+import format from 'date-fns/format'
 
 @Injectable()
 export class HealthDirectorateClientService {
-  constructor(private readonly api: StarfsleyfiAMinumSidumApi) {}
+  constructor(
+    private readonly starfsleyfiAMinumSidumApi: StarfsleyfiAMinumSidumApi,
+    private readonly vottordApi: VottordApi,
+  ) {}
 
-  private apiWithAuth = (user: User) =>
-    this.api.withMiddleware(new AuthMiddleware(user as Auth))
+  private starfsleyfiAMinumSidumApiWithAuth(auth: Auth) {
+    return this.starfsleyfiAMinumSidumApi.withMiddleware(
+      new AuthMiddleware(auth),
+    )
+  }
+
+  private vottordApiWithAuth(auth: Auth) {
+    return this.vottordApi.withMiddleware(new AuthMiddleware(auth))
+  }
 
   public async getHealthDirectorateLicense(
-    user: User,
+    auth: User,
   ): Promise<Array<MinarSidur> | null> {
-    return this.apiWithAuth(user).starfsleyfiAMinumSidumGet()
+    return this.starfsleyfiAMinumSidumApiWithAuth(
+      auth,
+    ).starfsleyfiAMinumSidumGet()
+  }
+
+  async getMyHealthcareLicenses(auth: Auth): Promise<HealthcareLicense[]> {
+    const items = await this.vottordApiWithAuth(
+      auth,
+    ).vottordStarfsleyfiVottordGet()
+
+    const result: HealthcareLicense[] = []
+
+    // loop through items to group together specialities per profession
+    for (let i = 0; i < items.length; i++) {
+      const item = items[i]
+      const professionId = item.idProfession
+
+      let index = result.findIndex((x) => x.professionId === professionId)
+
+      if (index === -1) {
+        const count = result.push({
+          professionId: item.idProfession || '',
+          professionNameIs: item.professionIsl || '',
+          professionNameEn: item.professionEn || '',
+          specialityList: [],
+          isTemporary: false,
+          isRestricted: item.isRestricted === 1,
+        })
+
+        index = count - 1
+      }
+
+      if (item.isSpeciality) {
+        result[index].specialityList.push({
+          specialityNameIs: item.specialityIsl || '',
+          specialityNameEn: item.specialityEn || '',
+        })
+      }
+
+      if (item.validTo) {
+        result[index].isTemporary = true
+        result[index].validTo = item.validTo
+      }
+    }
+
+    return result
+  }
+
+  async submitApplicationHealthcareLicenseCertificate(
+    auth: User,
+    request: HealthcareLicenseCertificateRequest,
+  ): Promise<HealthcareLicenseCertificate[]> {
+    const result: HealthcareLicenseCertificate[] = []
+
+    for (let i = 0; i < request.professionIdList.length; i++) {
```

Comment:
Can this be a long list of requests? 
There could be a benefit of a try catch to handle errors in the post requests if there are some errors in the for loop.

Reply:
Having an `await` in the `for`-loop seems a bitch sketchy, especially since the timeout is 30secs. The user might be waiting for 1 whole minute if they've selected 2 things, since the requests are sent out one after the other instead of in parallel. Can this be a `Promise.all` maybe, for concurrency?